.github/workflows/cloud-tpu-ci-nightly.yml

```
@@ -0,0 +1,61 @@
+name: Cloud TPU nightly
+
+on:
+  schedule:
+    - cron: "0 14 * * *"  # daily at 7am PST
+  workflow_dispatch:  # allows triggering the workflow run manually
+
+# Should also be set to read-only in the project settings.
+permissions:
+  contents: read
+
+jobs:
+  cloud-tpu-test:
+    runs-on: [self-hosted, tpu, v4-8]
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false  # don't cancel all jobs on failure
+      matrix:
+        python-version: ["3.10"]  # TODO(jakevdp): update to 3.11 when available.
+        jaxlib-version: [latest, nightly]
+    steps:
+      # https://opensource.google/documentation/reference/github/services#actions
+      # mandates using a specific commit for non-Google actions.
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@13ae5bb136fac2878aff31522b9efb785519f984
+        with:
+          python-version: ${{ matrix.python-version }}
```

You could cache the pip installs here

https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages

```suggestion
          python-version: ${{ matrix.python-version }}
          cache: 'pip'
          cache-dependency-path: build/test-requirements.txt
```

---

The GitHub actions cache. It actually might not work very well because you're not running in Azure right next to their caches